sess.go

```diff
@@ -92,8 +92,16 @@ func (s *session) Serve() error {
 		return err
 	}
 
+	s.initialized = true
+
 	// handle authentication.
-	err = s.Write(authOKMsg())
+	a := &authenticationNoPassword{}
+	authResponse, err := a.authenticate()
+	if err != nil {
+		return s.Write(errMsg(WithSeverity(err, "FATAL")))
```

Comment:
after revisiting authenticate code, i think it should return only error without ok/err message, and we should simply terminate the session in case of error.

the way i see it, only authenticator know if err msg should be sent.
for example, if authenticate gets error in rw.Read - no reason to write fatal error back to client.
instead, allow authenticate to send ok message and here only `return err` as any other case here...